test/functional/tests/fault_injection/test_fault_injection_many_to_one.py

```diff
@@ -214,10 +219,111 @@ def test_one_core_fail(cache_mode):
+          - Second core is able to use OpenCAS.
+    """
+    with TestRun.step("Prepare one cache and two core devices."):
+        cache_dev = TestRun.disks["cache"]
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)] * 2)
+        cache_part = cache_dev.partitions[0]
+        core_dev1 = TestRun.disks["core1"]  # This device would be unplugged.
+        core_dev1.create_partitions([Size(2, Unit.GibiByte)])
+        core_part1 = core_dev1.partitions[0]
+        core_dev2 = TestRun.disks["core2"]
+        core_dev2.create_partitions([Size(2, Unit.GibiByte)])
+        core_part2 = core_dev2.partitions[0]
+        Udev.disable()
+
+    with TestRun.step("Start cache"):
+        cache_mode = CacheMode.WB
+        cache = casadm.start_cache(cache_part, cache_mode=cache_mode, force=True)
+        caches_count = len(casadm_parser.get_caches())
+        if caches_count != 1:
+            TestRun.fail(f"Expected caches count: 1; Actual caches count: {caches_count}.")
+
+    with TestRun.step("Add both core devices to cache."):
+        core1 = cache.add_core(core_part1)
+        core2 = cache.add_core(core_part2)
+        cores_count = len(casadm_parser.get_cores(cache.cache_id))
+        if cores_count != 2:
+            TestRun.fail(f"Expected cores count: 2; Actual cores count: {cores_count}.")
+
+    with TestRun.step("Change sequential cutoff policy."):
+        cache.set_seq_cutoff_policy(SeqCutOffPolicy.never)
+
+    with TestRun.step("Fill cache with pages from the first core."):
+        blocks = int(cache.size / block_size.value)
+        dd = (
+            Dd()
+            .block_size(block_size)
+            .count(blocks)
+            .input("/dev/urandom")
+            .output(core1.path)
+            .oflag("direct")
+        ).run()
+        cache_occupied_blocks_before = cache.get_occupancy()
+        cache_dirty_blocks_before = cache.get_dirty_blocks()
+
+    with TestRun.step("Set cache mode"):
+        cache.set_cache_mode(cache_mode)
+
+    with TestRun.step("Unplug the first core device."):
+        core_dev1.unplug()
+
+    with TestRun.step("Check if core device is really out of cache."):
+        output = str(casadm.list_caches().stdout.splitlines())
+        if core_part1.path in output:
+            TestRun.fail("The first core device should be unplugged!")
+
+    with TestRun.step("Check if the remaining core is able to use cache."):
+        dd_builder(cache_mode, core2, Size(100, Unit.MebiByte)).run()
+        if not float(core2.get_occupancy().get_value()) == 0:
+            TestRun.LOGGER.error("Cache occupancy increased despite dirty data form first core!")
+        TestRun.LOGGER.info("The remaining core is not able to use cache.")
+
+    with TestRun.step("Check if occupancy from the first core is removed from cache."):
+        # Cache occupancy cannot be lower than before the first core fails and after that
+        # should be equal to the sum of occupancy of the first and the remaining core
```

Comment:
this comment doesn't make much sense anymore - see my next comment for explanatioin